Update delete_entries processor to add new features (#11281)
opensearch-project/documentation-website: _data-prepper/pipelines/configuration/processors/delete-entries.md

| `with_keys` | No       | An array that specifies the keys of the entries to delete.                                                                                                                                      |
| `with_keys_regex` | No       | An array of regular expression (regex) patterns used to match the keys of entries to delete.                                                                                                    |
| `exclude_from_delete` | No | A set of entries to exclude from deletion when using the `with_keys_regex` configuration.                                                                                                       |
| `entries` | No | A list of entries to delete from the event.                                                                                                                                                     |
| `delete_when` | No | Specifies under what conditions the deletion should be perfomed. (Example: <i>value= "/some_key == null"</i> - Deletion runs only if <i>some_key</i> is null or does not exist.)                |

> |`delete_from_element_when` | No | Specifies the condition needed to delete the key from each element of a list whe using `iterate_on`                                                                                             |

Reviewer: ```suggestion
|`delete_from_element_when` | No | Specifies the condition for deleting the key from each element of a list when using `iterate_on`.                                                                                             |
```

> | `iterate_on` | No | A string that specifies the key of the list of objects to iterate over and delete the keys specified in `with_keys`                                                                             

Reviewer: ```suggestion
| `iterate_on` | No | A string specifying the key of the list of objects to iterate over. The keys listed in `with_keys` will be deleted from each object.                                                                             
```
